src/backend/marsha/core/serializers/playlist.py

```diff
@@ -26,15 +31,40 @@ class Meta:
             "users",
         ]
 
-    def create(self, validated_data):
-        """Create the Playlist object with the passed data."""
-        return Playlist.objects.create(**validated_data)
+    portable_to = serializers.SerializerMethodField(read_only=False)
+
+    def get_portable_to(self, obj):
+        """Getter for portable_to attribute instead of PlaylistSerializer to prevent recursion."""
+        return obj.portable_to.values("id", "title")
+
+    def to_internal_value(self, data):
+        """
+        Validate portable_to.
+
+        Raise a ValidationError if portable_to contains current object playlist id
+        or an unknown playlist id.
+        """
+        value: OrderedDict = super().to_internal_value(data)
+        if "portable_to" in data.keys():
+            reachable_from_playlist_ids = data.get("portable_to")
+            for playlist_id in reachable_from_playlist_ids:
+                try:
+                    Playlist.objects.get(pk=playlist_id)
+                except (Playlist.DoesNotExist, ValidationError) as error:
+                    raise ValidationError(
+                        _("Playlist %(playlist_id)s does not exists.")
+                        % {"playlist_id": playlist_id}
+                    ) from error
+                if playlist_id == str(self.instance.id):
+                    raise ValidationError(_("Playlist is not portable to itself."))
+            value.update({"portable_to": reachable_from_playlist_ids})
+        return value
```

Comment:
my comment was lost when you pushed :sob: I was proposing something like this to solve:
- doing it in one query instead of N
- doing a light query instead of getting full objects with get

```suggestion
        if "portable_to" in data.keys():
            reachable_from_playlist_ids = data.get("portable_to")
            if str(self.instance.id) in reachable_from_playlist_ids:
                raise ValidationError(_("Playlist is not portable to itself."))
            existing_playlist_ids = Playlist.objects.filter(id__in=reachable_from_playlist_ids).values_list("id", flat=True)
            ids_diff = set(reachable_from_playlist_ids) - set(existing_playlist_ids)
            if ids_diff:
                raise ValidationError(
                    _("Some playlists don't exist: {ids_diff!s}.")
                )
            value.update({"portable_to": reachable_from_playlist_ids})
        return value
```